packages/server/modules/workspaces/repositories/projectRegions.ts

```diff
@@ -0,0 +1,218 @@
+import {
+  BranchCommits,
+  Branches,
+  Commits,
+  StreamCommits,
+  StreamFavorites,
+  Streams,
+  StreamsMeta
+} from '@/modules/core/dbSchema'
+import { Branch } from '@/modules/core/domain/branches/types'
+import { Commit } from '@/modules/core/domain/commits/types'
+import { Stream } from '@/modules/core/domain/streams/types'
+import {
+  BranchCommitRecord,
+  CommitRecord,
+  StreamCommitRecord,
+  StreamFavoriteRecord,
+  StreamRecord
+} from '@/modules/core/helpers/types'
+import { executeBatchedSelect } from '@/modules/shared/helpers/dbHelper'
+import {
+  CopyProjectModels,
+  CopyProjects,
+  CopyProjectVersions,
+  CopyWorkspace
+} from '@/modules/workspaces/domain/operations'
+import { WorkspaceNotFoundError } from '@/modules/workspaces/errors/workspace'
+import { Knex } from 'knex'
+import { Workspace } from '@/modules/workspacesCore/domain/types'
+import { Workspaces } from '@/modules/workspacesCore/helpers/db'
+
+const tables = {
+  workspaces: (db: Knex) => db<Workspace>(Workspaces.name),
+  projects: (db: Knex) => db<Stream>(Streams.name),
+  models: (db: Knex) => db<Branch>(Branches.name),
+  versions: (db: Knex) => db<Commit>(Commits.name),
+  branchCommits: (db: Knex) => db<BranchCommitRecord>(BranchCommits.name),
+  streamCommits: (db: Knex) => db<StreamCommitRecord>(StreamCommits.name),
+  streamFavorites: (db: Knex) => db<StreamFavoriteRecord>(StreamFavorites.name),
+  streamsMeta: (db: Knex) => db(StreamsMeta.name)
+}
+
+export const copyWorkspaceFactory =
+  (deps: { sourceDb: Knex; targetDb: Knex }): CopyWorkspace =>
+  async ({ workspaceId }) => {
+    const workspace = await tables
+      .workspaces(deps.sourceDb)
+      .select('*')
+      .where({ id: workspaceId })
+
+    if (!workspace) {
+      throw new WorkspaceNotFoundError()
+    }
+
+    await tables
+      .workspaces(deps.targetDb)
+      .insert(workspace)
+      .onConflict(Workspaces.withoutTablePrefix.col.id)
+      .merge(Workspaces.withoutTablePrefix.cols as (keyof Workspace)[])
+
+    return workspaceId
```

Comment:
we should check how workspaces are propagated to the region db and use that to publish the change:
- always update the main db
- update (fake) the main db and use the publish system to update the region

Reply:
It appears that we quite literally only care that the workspace id is defined in the regional database for foreign key constraints. Anything user-visible is pulled from the main db record. Simplified this to a copy then ignore on conflict.